src/docbuilder/rustwide_builder.rs

```diff
@@ -309,6 +310,26 @@ impl RustwideBuilder {
         }
 
         let limits = Limits::for_crate(&mut conn, name)?;
+        #[cfg(target_os = "linux")]
+        if !self.config.disable_memory_limit {
+            let mem_info = procfs::Meminfo::new().context("failed to read /proc/meminfo")?;
+            let available = mem_info.mem_available.unwrap_or(
+                // Assume about half of cached memory can be reclaimed without thrashing.
+                // This is a terrible hack.
+                mem_info.mem_free + mem_info.cached / 2,
```

Comment:
On my system at least `free + cached` is ~half of `available`, I think because of how ZFS is implemented it ends up being attributed as `k_reclaimable` instead of `cached`.

But, this branch should never be taken, `available` is 7 years old at this point, I'd check it's available in production then change this to `expect("Your kernel is too old")`.

Reply:
Yup, this is available in prod:
```
$ cat /proc/meminfo | grep -i available
MemAvailable:   14260052 kB
```
Thanks.